docs/snippets/gpu/preprocessing_single-plane.cpp

```diff
@@ -0,0 +1,55 @@
+#include <openvino/runtime/core.hpp>
+#define OV_GPU_USE_OPENCL_HPP
+#include <openvino/runtime/intel_gpu/ocl/ocl.hpp>
+#include <openvino/runtime/intel_gpu/properties.hpp>
+#include <openvino/core/preprocess/pre_post_process.hpp>
+
+ov::intel_gpu::ocl::ClImage2DTensor get_y_tensor();
+ov::intel_gpu::ocl::ClImage2DTensor get_uv_tensor();
+
+int main() {
+    ov::Core core;
+    auto model = core.read_model("model.xml");
+
+    //! [init_preproc]
+    using namespace ov::preprocess;
+    auto p = PrePostProcessor(model);
+    p.input().tensor().set_element_type(ov::element::u8)
+                      .set_ layout("NHWC")
+                      .set_memory_type(ov::intel_gpu::memory_type::surface);
+    p.input().model().set_layout("NCHW");
+    auto model_with_preproc = p.build();
+    //! [init_preproc]
+
+    auto compiled_model = core.compile_model(model, "GPU");
+    auto context = compiled_model.get_context().as<ov::intel_gpu::ocl::ClContext>();
+    auto input = model->get_parameters().at(0);
+    auto infer_request = compiled_model.create_infer_request();
+
+{
+    //! [single_batch]
+    cl::Image2D img_y_plane;
+    auto input_y = model_with_preproc->get_parameters().at(0); 
+    auto remote_y_tensor = remote_context.create_tensor(input_y->get_element_type(), in_shape, img_y_plane);
+    inf_req_remote.set_tensor(*input_y->output(0).get_tensor().get_names().begin(), remote_y_tensor);
```

Comment:
@vladimir-paramuzov 
That's a very good point. I have no expertise to improve the snippet, but if it is how you describe it, I assume we may want to reword the description also - lines 188/189. Do you think it would be proper to write:
calling the `create_tensor()` function creates one `ov::RemoteTensor` object, representing both the Y and UV planes, stored in that order;

Reply:
@vladimir-paramuzov , actually, it was an intention to describe exactly that case (where only one Y plane from two is used) for NV12 to Gray conversion. Should we call it something like "single_Y_plane" processing or something else to avoid misleading?

Reply:
@sshlyapn If the intention is to describe nv12->gray conversion, then it should be called nv12->gray, not single-plane preprocessing, as those are different use cases. Ideally, we need to describe both scenarios and cover them with snippets.

Reply:
That's all very interesting :) So am I correct in thinking that we can store the planes like so:
- both planes in two files
- both planes in one file
- one plane in one file

And the snippet presents the case with only Y stored in one object? If so, how about we change the description to something like:

* two-plane NV12 video surface input - calling the `create_tensor_nv12()` function creates
  a pair of `ov::RemoteTensor` objects, one per each plane;
* single-plane NV12 video surface input - calling the `create_tensor()` function creates one 
  `ov::RemoteTensor` object, representing either both planes together (Y elements 
  before UV elements) or just a single plane.

Would that justify the "single-plane" snippet?